src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/IServiceProviderExtensions.cs

```diff
@@ -15,5 +16,22 @@ internal static class IServiceProviderExtensions
         {
             return (InterfaceType)sp.GetService(typeof(ServiceType));
         }
+
+        public static IProjectService GetProjectService(this IServiceProvider sp)
+        {
+            var componentModel = sp.GetService(typeof(SComponentModel)) as IComponentModel;
+            if (componentModel == null)
+            {
+                return null;
+            }
+
+            var projectServiceAccessor = componentModel.GetService<IProjectServiceAccessor>();
```

Comment:
Seems like we can not - it is documented in CPS to get it this way

---
In reply to: [74310461](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74310461) [](ancestors = 74310461)